src/test/kotlin/nl/tudelft/ewi/se/ciselab/testgenie/uiTest/SettingsEvoSuiteConfigurableTest.kt

```diff
@@ -0,0 +1,273 @@
+        assertThat(oldValue).isNotEqualTo(settingsState.algorithm)
+        settingsComponent.algorithm = oldValue
+    }
+
+    fun testIsModifiedAlgorithmComboBox() {
+        val oldValue = settingsComponent.algorithm
+        settingsComponent.algorithm = ContentDigestAlgorithm.DYNAMOSA
+        assertThat(settingsConfigurable.isModified).isTrue
+        settingsComponent.algorithm = oldValue
+    }
+
+    fun testIsModifiedCheckBoxes() {
+        helperIsModifiedCheckBoxes(checkBoxGenerator())
+    }
+
+    fun testResetCheckBoxes() {
+        helperResetCheckBoxes(checkBoxGenerator())
+    }
+
+    fun testApplyCheckBoxes() {
+        helperApplyCheckBoxes(checkBoxGenerator())
+    }
+
+    /**
+     * Helper method to test if Reset is working correctly when modifying checkbox values.
+     *
+     * @param functions stream of expressions to execute
+     */
+    private fun helperResetCheckBoxes(functions: Stream<() -> Triple<() -> Unit, () -> Boolean, () -> Boolean>>) {
+        for (function in functions) {
+            val triple = function()
+            triple.first()
+            settingsConfigurable.reset()
+            assertThat(triple.second()).isTrue
+            assertThat(triple.third()).isTrue
+        }
+    }
+
+    /**
+     * Helper method to test if Apply is working correctly when modifying checkbox values.
+     *
+     * @param functions stream of expressions to execute
+     */
+    private fun helperApplyCheckBoxes(functions: Stream<() -> Triple<() -> Unit, () -> Boolean, () -> Boolean>>) {
+        for (function in functions) {
+            val triple = function()
+            triple.first()
+            settingsConfigurable.apply()
+            assertThat(triple.second()).isFalse
+            assertThat(triple.third()).isFalse
+        }
+    }
+
+    /**
+     * Helper method to test if isModified is working correctly when modifying checkbox values.
+     *
+     * @param functions stream of expressions to execute
+     */
+    private fun helperIsModifiedCheckBoxes(functions: Stream<() -> Triple<() -> Unit, () -> Boolean, () -> Boolean>>) {
+        for (function in functions) {
+            val triple = function()
+            triple.first()
+            assertThat(settingsConfigurable.isModified).isTrue
+            assertThat(triple.second()).isFalse
+            assertThat(triple.third()).isTrue
+        }
+    }
+
+    /**
+     * Helper method to generate the stream of functions.
+     *
+     * @return stream of functions
+     */
+    private fun checkBoxGenerator(): Stream<() -> Triple<() -> Unit, () -> Boolean, () -> Boolean>> {
+        return Stream.of(
+            {
+                val oldValue = settingsComponent.junitCheck
+                return@of Triple(
+                    { settingsComponent.junitCheck = !settingsComponent.junitCheck },
+                    { settingsComponent.junitCheck == oldValue },
+                    { settingsState.junitCheck == oldValue }
```

Comment:
Is it possible to return two functions that return only `{settingsComponent.jUnitCheck}` and `{settingsState.junitCheck}`, instead of three which do comparisons and have a lot of code duplication?

Reply:
Comparissons have been removed - good insight. Returning only 2 functions is not possible, because I need to keep track of previous value before updating it.